### ISSUE TITLE
OpenShift Web Console Incorrectly Reports "Failed" OperatorStatus

### ISSUE THREAD

**Issue by johnrk** [cockroachdb/cockroach-operator#436]
When installing a CrdbCluster on OpenShift Web Console, it incorrectly reports "Failed" OperatorStatus in the UI.

**Reproduction Steps**
1. On the OpenShift Web Console, Create a CrdbCluster with nodes=5, CockroachDB Version= v20.2.7
2. Immediately open the CockroachDB Overview page and observe the OperatorStatus field

**Expected Result:**
I would expect it to report that it is in progress or finished

**Actual Result:**
It communicates the OperatorStatus is "Failed"

[Loom Recording](https://www.loom.com/share/9c3d4d82495a4c718a558630a7d8d209)

**Comment by keith-mcclellan** [comment]
John, did you check to make sure all the pods actually started?  Just because you can see the service endpoint doesn't mean the database actually started - we'd need to see if there was an issue scheduling the pods being it.

**Comment by johnrk** [comment]
@keith-mcclellan , yes I did check. The pods eventually start. The issue here is receiving any kind of failure status (without explanation) when one is using the tool properly. As an operator, I would be alarmed.

**Comment by keith-mcclellan** [comment]
i agree this needs to be fixed if the DB did actually start, just trying to determine if the resource had actually failed or not! thanks.

@alinadonisa @chrislovecnm is there a way to set this status to pending instead of failed to start?

**Comment by alinadonisa** [comment]
@keith-mcclellan  the status logic is a bit more complex. We should be brainstorming more on this to make sure we have the correct behaviour.
